.github/workflows/build_go_library.yaml

```diff
@@ -182,63 +200,43 @@ jobs:
-            for engine in exporter_config:
-              os_type=exporter_config[engine]
-              ext='tar'
-              file=f'${{ inputs.exporter_name }}-{os_type}-{engine}'
-              print(file)
-              if '${{github.repository}}' == 'TessellDevelopment/tessell-db-perf-insights-exporter':
-                ext='tar.gz'
-              if '${{github.repository}}' == 'TessellDevelopment/tessell-victoria-metrics':
-                ext='tar.gz'
-                file=f'${{ inputs.exporter_name }}-{os_type}'
-                print(file)
-              print(\"{exporter_name}-{os_type}-{engine}\".format(os_type=exporter_config[engine], engine=engine,exporter_name=exporter_name )) 
-              exit_code = os.system(\"mkdir -p $HOME/.m2\")
-              if exit_code != 0:
-                raise Exception('Command failed with exit code {}'.format(exit_code))
-              exit_code = os.system(\"cp .github/scripts/settings.xml $HOME/.m2/settings.xml\")
-              if exit_code != 0:
-                raise Exception('Command failed with exit code {}'.format(exit_code))
-              exit_code = os.system(\"cat $HOME/.m2/settings.xml\")
-              if exit_code != 0:
-                raise Exception('Command failed with exit code {}'.format(exit_code))
-              exit_code = os.system(\"mvn deploy:deploy-file -Dnexus_url=https://${NEXUS_SERVER}/repository/${NEXUS_PULL_REPOS_M2} \
-                            -Dnexus_username=${GITHUB_USER} -Dnexus_password=${GITHUB_TOKEN} \
-                            -DgroupId=tessell.${{ inputs.exporter_name }} -DartifactId={file} -Dversion=${{ steps.latest_tag.outputs.tag_ref }}\
-                            -DgeneratePom=true -Dpackaging={ext} \
-                            -Durl=https://${NEXUS_SERVER}/repository/${NEXUS_PUSH_REPOS_M2} \
-                            -Dfile=./{file}.{ext} -DrepositoryId=nexus \".format(os_type=exporter_config[engine], engine=engine, ext=ext, file=file))
-              if exit_code != 0:
-                raise Exception('Command failed with exit code {}'.format(exit_code))
+            import yaml
+            with open('convoy.yaml', 'r') as yaml_file:
+              yaml_data = yaml.safe_load(yaml_file)
+            artifacts = yaml_data.get('generates').get('artifacts')
+            for artifact in artifacts:
+                file = artifact['name']
+                ext = artifact['extension']
+                print(file+'-'+ext)
+                exit_code = os.system(\"mkdir -p $HOME/.m2\")
+                if exit_code != 0:
+                  raise Exception('Command failed with exit code {}'.format(exit_code))
+                exit_code = os.system(\"cp .github/scripts/settings.xml $HOME/.m2/settings.xml\")
+                if exit_code != 0:
+                  raise Exception('Command failed with exit code {}'.format(exit_code))
+                exit_code = os.system(\"cat $HOME/.m2/settings.xml\")
```

Comment:
Use either `os.exists` and/or `os.stat` instead. The idea is to use language specific capabilities and not switching to `os.system` which will invoke another process; With python methods you will have better control over logic.